scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerStateRecoverHelper.java

```diff
@@ -129,15 +135,14 @@ private void recoverRunningTasksOrResetToPending(InternalJob job, List<InternalT
             if ((task.getStatus() == TaskStatus.RUNNING || task.getStatus() == TaskStatus.PAUSED) &&
                 task.getExecuterInformation() != null) {
                 try {
-                    TaskLauncher launcher = task.getExecuterInformation().getLauncher();
-                    logger.info("Recover running task " + task.getId() + " (" + task.getName() +
-                                ") successfully. Launcher: " + launcher);
-                    runningTasksCount++;
+                    if (runningTaskMustBeResetToPending(task, rmProxy)) {
+                        pendingTasksCount = setStatusAndIncrementPendingTasks(pendingTasksCount, task);
+                    } else {
+                        runningTasksCount++;
+                    }
                 } catch (Throwable e) {
-                    logger.info("Recover running task " + task.getId() + " (" + task.getName() +
-                                ") failed. Moving back task from status RUNNING TO PENDING.", e);
-                    task.setStatus(TaskStatus.PENDING);
-                    pendingTasksCount++;
+                    logger.warn("Fail to recover running task " + task.getId() + " (" + task.getName() + ")", e);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Fail to recover running task " 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)